### DISCUSSION TITLE
How to format row or column as percent

### DISCUSSION THREAD

**Discussion by mcp292** [vega/altair#3131] [Q&A] [answered]
# Problem
How do you format the title of row or column facets as percent?

For normal axes it's `alt.Y().axis(format="%")`, but `alt.Row().axis(format="%")` does not work, reporting `AttributeError: 'Row' object has no attribute 'axis'`.

# [MWE](https://altair-viz.github.io/gallery/bar_chart_trellis_compact.html)
```py
import altair as alt
import pandas as pd


source = pd.DataFrame(
    [
        {"a": "a1", "b": "b2", "c": "z", "p": "0.55"},
        {"a": "a1", "b": "b3", "c": "y", "p": "0.58"},
        {"a": "a2", "b": "b3", "c": "x", "p": "0.74"},
        {"a": "a2", "b": "b3", "c": "y", "p": "0.64"},
    ]
)

alt.Chart(source, width=60, height=alt.Step(8)).mark_bar().encode(
    alt.Y("c:N").axis(None),
    alt.X("a:N").title("Factor A"),
    alt.Color("c:N").title("settings").legend(orient="bottom", titleOrient="left"),
    alt.Row("p:Q"),
    # alt.Row("p:Q").axis(format="%"),
    alt.Column("b:N").title("Factor B"),
).save("test.png")
```
![image](https://github.com/altair-viz/altair/assets/49123398/7e279cc4-65e3-45f8-8b51-b724cefc93b9)

This graph makes no sense; I reached for the quickest example to modify.

**Comment by mcp292** [comment]
Bump.

**Comment by joelostblom** [comment, marked as answer]
You can use `header` instead of `axis`. Note that it seems like you also need to round the percentage so something like this would work `alt.Row("p:Q").header(format=".0%")`:

![image](https://github.com/altair-viz/altair/assets/4560057/b4ba29be-50fb-474e-a6a3-a018b8b860be)

**Reply by mcp292** [reply]
Thank you so much!

This works but still some weird behavior I'm not fully understanding.

1. Why does `format` need a precision specifier? With `alt.X().axis(format="%")` this is not needed. Why does the [`header`](https://altair-viz.github.io/user_guide/generated/core/altair.Header.html#altair.Header) `format` behavior differ from the [`axis`](https://altair-viz.github.io/user_guide/generated/core/altair.Axis.html#altair.Axis) `format` behavior when the documentation doesn't?
2. Why do I get the following warning: `WARN row encoding should be discrete (ordinal / nominal / binned).`?

**Reply by joelostblom** [reply]
1. I am not sure, that was suprising to me too. I guess there are different default formatting rules between axes and headers.
2. That is because Vega-Lite recommends to use one of those types for the row encoding. You  can specify it with `p:N` (and still do the formatting). I don't think anything will fail with quantitative, but not sure

**Reply by mcp292** [reply]
Thank you!
